common/snapshots.py

```diff
@@ -783,183 +783,186 @@ def backup(self, force = False):
 
                 # global flock to block backups from other profiles or users
                 # (and run them serialized)
-                self.flockExclusive()
-                logger.info('Lock', self)
+                # self.flockExclusive()
+                with flock.GlobalFlock():
+                    logger.info('Lock', self)
 
-                now = datetime.datetime.today()
+                    now = datetime.datetime.today()
 
-                # inhibit suspend/hibernate during snapshot is running
-                self.config.inhibitCookie \
-                    = tools.inhibitSuspend(toplevel_xid=self.config.xWindowId)
+                    # inhibit suspend/hibernate during snapshot is running
+                    self.config.inhibitCookie \
+                        = tools.inhibitSuspend(toplevel_xid=self.config.xWindowId)
 
-                # mount
-                try:
-                    hash_id = mount.Mount(cfg = self.config).mount()
+                    # mount
+                    try:
+                        hash_id = mount.Mount(cfg = self.config).mount()
 
-                except MountException as ex:
-                    logger.error(str(ex), self)
-                    instance.exitApplication()
-                    logger.info('Unlock', self)
-                    time.sleep(2)
+                    except MountException as ex:
+                        logger.error(str(ex), self)
+                        instance.exitApplication()
+                        logger.info('Unlock', self)
+                        time.sleep(2)
 
-                    return True
+                        return True
```

Comment:
This might have caused bugs in the past. The flock was never released when this "return" took affect. Now with the context manager the flock is always released.